src/interpreted/parser.py

```diff
@@ -302,6 +313,91 @@ def parse_single_line_statement(
         self.expect(TokenType.NEWLINE, TokenType.EOF)
         return node
 
+    def parse_import(self) -> Import:
+        self.expect(TokenType.NAME)
+
+        names = []
+        module = self.current().string
+
+        while True:
+            if self.match_op(","):
+                self.expect(TokenType.NAME)
+                names.append(alias(name=module, asname=None))
+                module = self.current().string
+                continue
+
+            elif self.match_name("as"):
+                self.expect(TokenType.NAME)
+                alias_name = self.current().string
+                names.append(alias(name=module, asname=alias_name))
+                if self.peek().token_type in (TokenType.NEWLINE, TokenType.EOF):
+                    break
+                elif self.match_op(","):
+                    self.expect(TokenType.NAME)
+                    module = self.current().string
+                continue
+
+            if self.match_op("."):
+                self.expect(TokenType.NAME)
+                module += "." + self.current().string
+                continue
+
+            if self.peek().token_type in (TokenType.NEWLINE, TokenType.EOF):
+                names.append(alias(name=module, asname=None))
+                break
+
+        return Import(names=names)
+
+    def parse_importfrom(self) -> ImportFrom:
+        self.expect(TokenType.NAME)
+
+        module_name = self.current().string
+        names = []
+
+        # parse submodule names or direct import keyword
+        if self.match_op("."):
+            while not self.match_name("import"):
+                if self.match_op("."):
+                    continue
+                self.expect(TokenType.NAME)
+                module_name += "." + self.current().string
```

Comment:
Small problem here, `from a.....b import x` will be valid because of this, and the token will output `module = 'a.b'`